streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -1095,9 +1080,6 @@ private <VTable, VOut> KStream<K, VOut> doStreamTableJoin(final KTable<K, VTable
                                                               final ValueJoinerWithKey<? super K, ? super V, ? super VTable, ? extends VOut> joiner,
                                                               final JoinedInternal<K, V, VTable> joinedInternal,
                                                               final boolean leftJoin) {
-        Objects.requireNonNull(table, "table can't be null");
-        Objects.requireNonNull(joiner, "joiner can't be null");
```

Comment:
Unnecessary -- this is checked in callee method already (and we need to keep it in the callees and cannot pull it in here)